src/System.Management.Automation/engine/remoting/common/RemoteSessionNamedPipe.cs

```diff
@@ -1320,19 +1295,22 @@ protected override NamedPipeClientStream DoConnect(int timeout)
                 }
             } while (elapsedTime < timeout);
 
+            SafePipeHandle pipeHandle = null;
             try
             {
+                pipeHandle = new SafePipeHandle(handle, ownsHandle: true);
```

Comment:
If we set ownsHandle to false this will block dispose in NamedPipeClientStream - who will close the handle?

https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/SafeHandle.cs,237

Reply:
Oh my apologies, the Dispose() call is in a catch block not finally. I thought it was a finally block so this is all good.